python/ngen_cal/src/ngen/cal/_hookspec.py

```diff
@@ -42,3 +44,21 @@ def ngen_cal_finish(exception: Exception | None) -> None:
     raised during the calibration loop.
     `exception` will be non-none if an exception was raised during calibration.
     """
+
+class ModelHooks():
+    @hookspec(firstresult=True)
+    def ngen_cal_model_output(id: str | None) -> Series:
+        """
+            Called during each calibration iteration to provide the model output in the form
+            of a pandas Series, indexed by time.
+            Output series should be in units of cubic meters per second.
+        """
+
+    @hookspec
+    def ngen_cal_model_post_iteration(path: Path, iteration: int) -> None:
```

Comment:
What about these names instead?
```python
    def ngen_cal_iteration_trailing(path: Path, iteration: int) -> None: ...
    def ngen_cal_iteration_finish(path: Path, iteration: int) -> None: ...
    def ngen_cal_iteration_end(path: Path, iteration: int) -> None:
```

Reply:
I don't know why, but the `path` argument feels a little weird. It feels like its probably helpful in our current use cases, but I have a feeling there are other fields that could be useful. We might consider creating a [`typing.TypedDict`](https://docs.python.org/3/library/typing.html#typing.TypedDict) that gets passed in here and has a `path` field plus other things that we can add in the future.

Reply:
Instead we could just pass a `JobMeta` here that has the `path`. I am a fan of that. I wouldnt be opposed to moving `JobMeta` to a `typing.TypedDict` either.

Reply:
What about `ngen_cal_iteration_finish(iteration: int, job_meta: JobMeta)`?

Reply:
renamed and adjusted to use job meta in 0b272d088de21ca23e6c35ff804a7aa004e3da56